plugins/imu/src/IMU.cc

```diff
@@ -57,18 +75,92 @@ void GazeboYarpIMU::Load(sensors::SensorPtr _sensor, sdf::ElementPtr _sdf)
     if (!configuration_loaded) {
         return;
     }
-    m_sensorName = _sensor->ScopedName();
+    m_scopedSensorName = _sensor->ScopedName();
 
     //Insert the pointer in the singleton handler for retriving it in the yarp driver
     GazeboYarpPlugins::Handler::getHandler()->setSensor(_sensor.get());
 
-    m_parameters.put(YarpIMUScopedName.c_str(), m_sensorName.c_str());
+    /*
+    * Open the driver wrapper
+    */
+    //Retrieve wrapper properties
+    ::yarp::os::Bottle MASwrapperProp = m_parameters.findGroup("WRAPPER");
+    if(MASwrapperProp.isNull())
+    {
+        yWarning("GazeboYarpIMU : [WRAPPER] group not found in config file, maybe you are using the old version of the ini file, please update icub-gazebo\n");
+        yWarning("GazeboYarpIMU : trying to open it with the legacy behaviour device-subdevice");
+        legacy_behaviour = true;
+    }
+
+    if (legacy_behaviour) {
+        m_parameters.put(YarpIMUScopedName.c_str(), m_scopedSensorName.c_str());
+        m_parameters.put("sensor_name",_sensor->Name());
+        if (!m_imuDriver.open(m_parameters)) {
+            yError() << "GazeboYarpIMU Plugin Load failed: error in opening yarp driver";
+        }
+        return;
```

Comment:
In this case, I assume we are inducing a wrong behavior, by not attaching itself to the analog server?

Reply:
Actually inside this if statement there is the code as it was before, then if you have the old ini file it will open as `device` the `ServerInertial` and as `subdevice` the `GazeboYarpIMU`. No attach is required in this case as before.

Reply:
Ah okay..! Understood, since we are dealing with `ServerInertial`. Thanks @Nicogene !